qiskit_experiments/test/mock_iq_helpers.py

```diff
@@ -0,0 +1,394 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Probability and phase functions for the mock IQ backend."""
+
+from abc import abstractmethod
+from typing import Dict, List
+import numpy as np
+from qiskit import QuantumCircuit
+from qiskit.providers.aer import AerSimulator
+
+
+class MockIQExperimentHelper:
+    """Abstract class for the MockIQ helper classes"""
+
+    @abstractmethod
+    def compute_probabilities(self, circuits: List[QuantumCircuit]) -> List[Dict[str, float]]:
+        """
+        A function provided by the user which is used to determine the probability of each output of the
+         circuit. The function returns a list of dictionaries, each containing output binary strings and
+          their probabilities.
+
+        Examples:
+
+            **1 qubit circuit - excited state**
+
+            In this experiment, we want to bring a qubit to its excited state and measure it.
+            The circuit:
+                         ┌───┐┌─┐
+                      q: ┤ X ├┤M├
+                         └───┘└╥┘
+                    c: 1/══════╩═
+                               0
+
+            The function that calculates the probability for this circuit, doesn't need any
+            calculation_parameters. It will be as following:
+
+            .. code-block::
+
+                @staticmethod
+                def compute_probabilities(self, circuits: List[QuantumCircuit])
+                    -> List[Dict[str, float]]:
+
+                    output_dict_list = []
+                    for circuit in circuits:
+                        probability_output_dict = {"1": 1, "0": 0}
```

Comment:
```suggestion
                        probability_output_dict = {"1": 1.0, "0": 0.0}
```
I would add the decimals here to be explicit that these are not counts.

Reply:
Changed as suggested.